scrapers_next/ma/committees.py

```diff
@@ -0,0 +1,89 @@
+from spatula import URL, HtmlListPage, HtmlPage, CSS, XPath, selectors
+from openstates.models import ScrapeCommittee
+import requests
+
+requests.packages.urllib3.disable_warnings()
+
+
+class CommitteeDetail(HtmlPage):
+    def process_page(self):
+        com = self.input
+        com.add_source(
+            self.source.url,
+            note="Committee Details Page",
+        )
+        try:
+            member_blocks = XPath('//*[contains(@class, "committeeMemberList")]').match(
+                self.root
+            )[0]
+        except selectors.SelectorError:
+            return com
+        for data in member_blocks:
+            # break if no members is in the list
+            if "There are no members" in data.text:
+                break
+            # continue to extract members if available
+            name = None
+            role = "Member"
+            for info in data.text_content().splitlines():
+                if not info.strip():
+                    continue
+                if not name:
+                    name = info.strip().title()
+                else:
+                    role = info.strip().replace(")", "").replace("(", "").title()
+            com.add_member(name, role)
+
+        return com
+
+
+class CommitteeList(HtmlListPage):
+    source = URL(
+        "https://malegislature.gov/Committees",
+        headers={
+            "User-Agent": "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:108.0) Gecko/20100101 Firefox/108.0",
+            "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8",
+            "Accept-Language": "en-US,en;q=0.5",
+            "Referer": "https://www.google.com/",
+        },
+        verify=False,
+    )
+
+    def process_page(self):
+        for committees in XPath('//*[contains(@class, "committeeList")]').match(
+            self.root
+        ):
+            for each_committee in committees:
+                name = str(each_committee.text_content())
+                if any(name.startswith(s) for s in ["House", "Senate", "Joint"]):
+                    cm_url = CSS("a").match_one(each_committee).get("href")
+                    if "house committee" in name.lower():
```

Comment:
While Massachusetts does not currently list subcommittees, we'd like the scraper to be ready to alert us if a subcommittee appears in the future (ideally we would write handling for it now, but we don't know how it would appear). Above line 60 (i.e. where you start checking for "house committee" etc.) let's add a check for the word "subcommittee" in `name.lower()`, and raise a custom exception in the event that it is.

Can you create that custom exception at the top of the file (just below your `disable_warnings()` call) that reads:
```
class SubcommitteeFound(BaseException):
    def __init__(self, com_name):
        super().__init__(f"Scraper has no process for ingesting subcommittee classification: {com_name}")
```

Reply:
added SubcommitteeFound